koku/api/query_params.py

```diff
@@ -181,6 +185,36 @@ def _process_tag_query_params(self, query_params):
                 param_tag_keys.add(key)
         return param_tag_keys
 
+    # TODO: This function will be needed for below
+    # def _strip_aws_category_prefix(self, value):
+    #     if AWS_CATEGORY_PREFIX.replace(":", "") not in value:
+    #         return value
+    #     if value.startswith(AWS_CATEGORY_PREFIX):
+    #         return value[len(AWS_CATEGORY_PREFIX) :]  # noqa: E203
+    #     if value.startswith(AND_AWS_CATEGORY_PREFIX):
+    #         return value[len(AND_AWS_CATEGORY_PREFIX) :]  # noqa: E203
+    #     if value.startswith(OR_AWS_CATEGORY_PREFIX):
+    #         return value[len(OR_AWS_CATEGORY_PREFIX) :]  # noqa: E203
+
+    def _process_aws_category_query_params(self, query_params):
+        """Reduce the set of aws categories based on those being queried."""
+        param_aws_category_keys = set()
+        for key, value in query_params.items():
+            if not isinstance(value, (dict, list)):
+                value = [value]
+            for inner_key in value:
+                if AWS_CATEGORY_PREFIX in inner_key:
+                    if self.query_handler.provider in [Provider.PROVIDER_AWS, Provider.OCP_AWS]:
+                        param_aws_category_keys.add(inner_key)
+                # TODO: Remove when we have better serializer logic
+                # for aws_category keys
+                # stripped_key = self._strip_aws_category_prefix(inner_key)
+                # if stripped_key in self.aws_category_keys:
+            # stripped_key = self._strip_aws_category_prefix(key)
+            # if stripped_key in self.aws_category_keys:
+            #     param_aws_category_keys.add(key)
```

Comment:
```
It is not going to completed during this issue, I talked to Andrew this is an MVP and for some internal stakeholders. We are going to build the right urls for a customer and add them to an email we control as a quickest win scenario. Then we will go back and create a table that has the acceptable categories since the categories are created by the user it will work very similarly to what we do for tags in the reporting_awstags_summary table. However, that will be in a separate PR, and once we have that we can figure out if a category is acceptable as a parameter. So, these todos are to make it easier for when we come back to it later.
```